### ISSUE TITLE
Shader interpreter causing a crash on AMD GPU

### ISSUE THREAD

**Issue by Jacoby1218** [RPCS3/rpcs3#16374]
### Quick summary

When the options Async + Shader interpreter, or Shader interpreter only are used, RPCS3 crashes and closes. If the shader cache is partially built for a game, the first time the interpreter is called, RPCS3 will crash.

### Attach a log file

[RPCS3.log](https://github.com/user-attachments/files/18035084/RPCS3.log)

### System configuration

Windows 11 24H2
AMD Ryzen 7 5700X3D
AMD Radeon RX 7900 XT (driver version 24.10.1)

**Comment by Valtekken** [comment]
> This issue is still present on 24.12.1 (which came out literally minutes after me writing this issue)

It was actually released yesterday https://videocardz.com/driver/amd-radeon-software-adrenalin-24-12-1

**Comment by Jacoby1218** [comment]
This crash wasn't fixed by #16540, still happens in latest master

**Comment by kd-11** [comment]
Need to validate the shader and send the spv off to amd to fix the compiler if it's broken.

**Comment by kd-11** [comment]
Visual glitches are fine and expected. 99% of them are because we do not emulate the RSX weird floating point behavior in interpreter mode for performance reasons.
The shader compiler crashes are annoying though. I've seen this a few times when debugging.
The interpreter shader also takes like a full minute to build while its only a few seconds on NV drivers.
NV vulkan driver used to be pretty bad but they overhauled their shader compiler a few years ago and bow it's pretty good. No crashes so far (from shader compiler, device lost mess is still a problem with NV)

**Comment by Jacoby1218** [comment]
> Are there specific games to trigger the crash?

No, I can reliably trigger it in any game I try. RPCS3 just closes as soon as the shader interpreter is used. If you already have a shader cache for a game, you'd have to clear it before you'd see this crash (or get to a part of the game that forces you to compile them).

**Comment by Darkhost1999** [comment]
When I looked at the log to see what games might be tested and if I share any of the games I noticed you're using the deprecated path dev_hdd0/disc that can create its own problems. I sadly don't share any mutual game titles to test with 1 I observe you own.

I did test with clean shader cache as using Nvidia cache with my AMD igpu will crash and is unsupported at this time. See other issue/task.

Can you while answering if you can reproduce the issue consistently 100% of the time also reupload a log utilizing the new updated path rpcs3/games? Seeing if maybe path isn't creating an issue.
I also encourage validating the game files and ensuring they're not corrupted if you haven't done so before. There are instructions on our discord, forums, and the wiki page on how to do so, also should be a hyperlink in the template prior to hitting submit to create a new issue ticket here on GitHub.

**Comment by kd-11** [comment]
Ok, I just tried again, shader interpreter only, RDNA3 all caches deleted and it started happening reliably at boot. I'll run this through spirv-val, let's see if there's any bugs in the shader or if I just need to file this with AMD.

**Comment by kd-11** [comment]
Hmm, looks like it's our bug. I need to reimplement the binding slots stuff a bit.

**Comment by kd-11** [comment]
Should be fixed by https://github.com/RPCS3/rpcs3/pull/16576
Test using DeS on RDNA3 (you can see broken shadow caused by interpreter):

![Image](https://github.com/user-attachments/assets/424e5705-f57a-44c0-bd8a-a6dabe2fd7f5)

I also fixed the black textures in most games when using interpreter. That was a regression.
